src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/SignedXmlDebugLog.cs

```diff
@@ -571,7 +571,7 @@ internal static void LogNamespacePropagation(SignedXml signedXml, XmlNodeList na
         /// <param name="reference">The reference being processed</param>
         /// <param name="data">Stream containing the output of the reference</param>
         /// <returns>Stream containing the output of the reference</returns>
-        internal static Stream LogReferenceData(Reference reference, Stream data)
+        internal static Stream? LogReferenceData(Reference reference, Stream? data)
```

Comment:
I wonder if doing something like

```suggestion
        [return: NotNullIfNotNull("data")]
        internal static Stream? LogReferenceData(Reference reference, Stream? data)
```

will fix some unnecessary may-be-null states.

And since we'd NullReferenceException if `data` was null, maybe just getting rid of the `?`s here altogether is better.

Reply:
Thank you!